pkg/volume/csi/csi_block.go

```diff
@@ -252,22 +296,71 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	}
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
-	// Call NodePublishVolume
-	publishPath, err := m.publishVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment, stagingPath)
+	// In SetUpDevice, NodeStageVolume is called, but NodePublishVolume isn't,
+	// therefore, there will be no device available in publishPath, yet.
+	// However, GenerateMapVolumeFunc expects SetUpDevice to return the path that the device
```

Comment:
this comment feels off - if the interface is expecting something that we can't provide, that feels a lot like we're putting logic in the wrong place.